phoenicis-javafx/src/main/java/org/phoenicis/javafx/components/library/skin/LibraryDetailsPanelSkin.java

```diff
@@ -0,0 +1,214 @@
+     */
+    private GridPane createControlButtons() {
+        final GridPane controlButtons = new GridPane();
+        controlButtons.getStyleClass().add("shortcut-control-button-group");
+
+        ColumnConstraints runColumn = new ColumnConstraintsWithPercentage(33.3);
+        ColumnConstraints stopColumn = new ColumnConstraintsWithPercentage(33.3);
+        ColumnConstraints uninstallColumn = new ColumnConstraintsWithPercentage(33.3);
+
+        controlButtons.getColumnConstraints().addAll(runColumn, stopColumn, uninstallColumn);
+
+        final Button runButton = new Button(tr("Run"));
+        runButton.getStyleClass().addAll("shortcutButton", "runButton");
+        runButton.setOnMouseClicked(event -> Optional.ofNullable(getControl().getOnShortcutRun())
+                .ifPresent(onShortcutRun -> onShortcutRun.accept(getControl().getShortcut())));
+        GridPane.setHalignment(runButton, HPos.CENTER);
+
+        final Button stopButton = new Button(tr("Close"));
+        stopButton.getStyleClass().addAll("shortcutButton", "stopButton");
+        stopButton.setOnMouseClicked(event -> Optional.ofNullable(getControl().getOnShortcutStop())
+                .ifPresent(onShortcutStop -> onShortcutStop.accept(getControl().getShortcut())));
+        GridPane.setHalignment(stopButton, HPos.CENTER);
+
+        final Button uninstallButton = new Button(tr("Uninstall"));
+        uninstallButton.getStyleClass().addAll("shortcutButton", "uninstallButton");
+        uninstallButton.setOnMouseClicked(event -> Optional.ofNullable(getControl().getOnShortcutUninstall())
+                .ifPresent(onShortcutUninstall -> onShortcutUninstall.accept(getControl().getShortcut())));
+        GridPane.setHalignment(uninstallButton, HPos.CENTER);
+
+        controlButtons.addRow(0, runButton, stopButton, uninstallButton);
+
+        return controlButtons;
+    }
+
+    /**
+     * Updates the shortcutProperties of the shortcut in the given {@link GridPane propertiesGrid}
+     *
+     * @param propertiesGrid The shortcutProperties grid
+     */
+    private void updateProperties(final GridPane propertiesGrid) {
+        propertiesGrid.getChildren().clear();
+
+        for (Map.Entry<String, Object> entry : shortcutProperties.entrySet()) {
+            final int row = propertiesGrid.getRowCount();
+
+            final Label keyLabel = new Label(tr(decamelize(entry.getKey())) + ":");
+            keyLabel.getStyleClass().add("captionTitle");
+            GridPane.setValignment(keyLabel, VPos.TOP);
+
+            final Label valueLabel = new Label(entry.getValue().toString());
+            valueLabel.setWrapText(true);
+
+            propertiesGrid.addRow(row, keyLabel, valueLabel);
+        }
+    }
+
+    /**
+     * Updates the shortcut of this {@link LibraryDetailsPanelSkin} instance
+     */
+    private void updateShortcut() {
+        final ShortcutDTO shortcut = getControl().getShortcut();
```

Comment:
Shouldn't the control "control" the skin instead of the skin querying the control?

Reply:
No, the names chosen by JavaFX make it really difficult to understand what class is responsible for what:
- the control is the model, it contains the input and output fields of the component
- the skin is the view, it is responsible for taking the input fields from the component and displays its content
- the behavior is the controller, it is responsible for mapping changes done in the view to the model

Reply:
I see. That's a strange naming indeed.